app/models/product_drive.rb

```diff
@@ -69,6 +71,13 @@ def self.search_date_range(dates)
     @search_date_range = { start_date: dates[0], end_date: dates[1] }
   end
 
+  def validate_destroy
```

Comment:
This seems a bit heavy for a validation. I'd rather extract this to a `ProductDriveDestroyService` which calls the policy. That matches the pattern we have a bit better. It can be pretty lightweight.